lib/events/encryptedevent.cpp

```diff
@@ -0,0 +1,41 @@
+#include "encryptedevent.h"
+#include "room.h"
+
+#include <message.h> // QtOlm
+
+using namespace QMatrixClient;
+using namespace QtOlm;
+
+EncryptedEvent::EncryptedEvent(const Room *roomContext, const QJsonObject &ciphertext, const QString &senderKey)
```

Comment:
As discussed in Matrix, this does not bode well. Events shouldn't depend on `Room`, it can only be the other way around. (And one day I'll draft a high-level design document that describes that...) As also agreed, for now probably just provide `EncryptedEvent` as a tiny wrapper that doesn't store `_decryptedEvent` at all, and invoke decryption outside of the library. This and the next constructor then will not need `Room` to fulfil its job.